floodviz/static/js/hydrograph.js

```diff
@@ -48,35 +56,55 @@
 		// Set the ranges
 		var x = d3.scaleTime().range([0, width]);
 		var y = d3.scaleLog().range([height, 0]);
-		
-    /**
-			* @param {Array} data -- an array containing the json data to be drawn
-			*
-			* Draws the svg, scales the range of the data, and draws the line for each site
-			* all based on the data set as it was passed in. Called as needed
-			* when data changes (as in removal of a line).
-			*
-		*/
-		var update = function(data) {
+
+		/**
+		 * Filters a set of data based on the ids listed in display_ids
+		 * @param data The dataset to be filtered
+		 * @returns {Array} The entries of the original `data` whose `key` values are elements of display_ids.
+		 */
+		var subset_data = function (data) {
```

Comment:
Consider using the array filter function https://www.w3schools.com/jsref/jsref_filter.asp.